tony-core/src/main/java/com/linkedin/tony/util/Utils.java

```diff
@@ -410,7 +410,9 @@ public static void addEnvironmentForResource(LocalResource resource, FileSystem
   public static AMRMClient.ContainerRequest setupContainerRequestForRM(JobContainerRequest request) {
     Priority priority = Priority.newInstance(request.getPriority());
     Resource capability = Resource.newInstance((int) request.getMemory(), request.getVCores());
-    Utils.setCapabilityGPU(capability, request.getGPU());
+    if (request.getGPU() > 0) {
+      Utils.setCapabilityGPU(capability, request.getGPU());
```

Comment:
What happens with the prior behavior(Utils.setCapabilityGPU(capability, 0)?

Reply:
The function just returns with no harm.